proposals/p5913.md

```diff
@@ -807,57 +846,48 @@ As with the overall strict model, we suggest these rules to be considered
 state whenever there is a reasonable implementation strategy.
 
-We also specifically expect pointers and bool to implement IsUnformed, allowing
-types that build on top of these to use them to model their own unformed state.
+We also specifically expect pointers and bool to implement `IsUnformed`,
+allowing types that build on top of these to use them to model their own
+unformed state.
 
-## Class types with a vtable
+### Class types with a vtable
 
 **Future work:** it would be very nice to allow types to reuse the
 vtable-pointer field to implement their unformed state. This is left as future
 work to address how to allow types to control opting-in and opting-out of this
 behavior, as well as how it should work across inheritance.
 
-## Subsetting the object for its unformed state
-
-The idea of explicitly selecting a subset of the object that participates in its
-unformed state is important to dig into. The goal here is to enable types to
-define potentially large portions of their state as remaining uninitialized for
-efficiency without exposing code manipulating these types to initialization
-unsafety. Making the subset explicit is important both to allow checking that
-code is correctly accessing only the safe subset when operating on an unformed
-object, and to enable sanitizers style diagnostics for the rest of the object.
-
-### Subsetting by fields
-
-We propose subsetting by fields rather than by a more complex approach such as a
-[bit-mask](#bit-mask-based-unformed-state) because this is expected to be easy
-to implement at low overhead, and meshes nicely with the struct literal syntax
-already present in Carbon. This seems like an effective starting position, but
-we should revisit it when introducing bitfields more fully to the language and
-determining how to reason about them.
-
-## Comparison to `MaybeUninit` from Rust
+### Comparison to `MaybeUninit` from Rust
 
 It's important to compare and contrast what this proposal suggests to Rust's
 `MaybeUninit` as they can seem superficially very similar.
 
 Fundamentally, the goal of modeling unformed state is somewhat different from
-`MaybeUninit` -- it is about providing access to types' internal invalid or
-no-op states in order to provide improved ergonomics around initialization
-without a significant safety loss. It is very much focused on enabling _type
-design_ to opt into this, rather than intended for user code dealing with
-complex initialization. Carbon may well end up needing something like
-`MaybeUninit` to manage cases which cannot be modeled safely. Currently, the
-plan is to directly use raw storage for this, but if doing so surfaces an
-important abstraction layer on top like `MaybeUninit`, we should add it.
+`MaybeUninit` -- unformed state is about providing access to types' internal
+invalid or no-op states in order to provide improved ergonomics around
+initialization without a significant safety loss. It is very much focused on
```

Comment:
```suggestion
initialization without a significant safety loss. It is very much focused on
```
The goal isn't to have slightly less safety either. I think this is trying to say that you usually don't have to use `unsafe`?

Reply:
Yes, changed the wording here to talk about not needing unsafe user code.